libs/wingsdk/src/target-tf-azure/bucket.inflight.ts

```diff
@@ -0,0 +1,122 @@
+import { DefaultAzureCredential } from "@azure/identity";
+import { BlobServiceClient } from "@azure/storage-blob";
+import { BucketDeleteOptions, IBucketClient } from "../cloud";
+
+export class BucketClient implements IBucketClient {
+  private readonly bucketName: string;
+  private readonly blobServiceClient: BlobServiceClient;
+  private readonly defaultAzureCredential: DefaultAzureCredential =
+    new DefaultAzureCredential();
+
+  constructor(bucketName: string, storageAccount: string) {
```

Comment:
If you run into issues trying to unit test this, one strategy you could try is adding an optional third parameter for passing a blobServiceClient instance. In the jest test, you would pass in a jest mock object that just implements or stubs out the methods you need. (Maybe there's a more elegant way to do this? 🤷‍♂️)

(only suggesting because I know jest can be a pain sometimes... ping me if you have any questions 😄)